### DISCUSSION TITLE
Lazy loading of jupyter labextensions

### DISCUSSION THREAD

**Discussion by Raghav-254** [jupyterlab/jupyterlab#15945] [Q&A] [answered]
Hello community,

I’m currently using JupyterLab version 3.6.3 and I have a question regarding the lazy loading of extensions. I would like to know if the lazy loading feature is supported in this version. Specifically, I’m curious whether the JavaScript files associated with Jupyter labextensions are loaded only when the extension is in use, or if all the JS files associated with the labextensions are re-loaded on a jupyterlab page reload.

Also I observed that the lazy loading is not working when installing the extension via jupyter labextension install command (as a source extension) and working on installing the extension via pip install. Is this the desired behaviour or this obversion is incorrect.

Can you please answer these questions?

Also to experiment with I was installing this open source jupyterlab-drawio labextension. [GitHub - QuantStack/jupyterlab-drawio: A standalone embedding of the FOSS drawio / mxgraph package into jupyterlab](https://github.com/QuantStack/jupyterlab-drawio)

Thank you in advance for your assistance!

**Comment by krassowski** [comment, marked as answer]
> Specifically, I’m curious whether the JavaScript files associated with Jupyter labextensions are loaded only when the extension is in use, or if all the JS files associated with the labextensions are re-loaded on a jupyterlab page reload.

1. In JupyterLab 4.0 things were improved with respect to not pulling static assets for disabled extensions
2. Extension can use async imports to delay fetching things
3. Installing via `jupyter labextension install` is not recommended for users (in fact derpecated) and indeed will result in different bundling and serving of assets (in way which depends on what specific Node dependencies, importantly webpack, you have installed locally).

> JupyterLab version 3.6.3

FYI this will reach EOL relatively soon https://github.com/jupyterlab/jupyterlab/issues/15921
